client/me/purchases/manage-purchase/index.tsx

```diff
@@ -638,6 +639,49 @@ class ManagePurchase extends Component<
 		return null;
 	}
 
+	renderRefundText() {
+		const { purchase, translate } = this.props;
+
+		// Hide if refund window has lapsed.
+		if ( ! hasAmountAvailableToRefund( purchase ) || ! purchase?.mostRecentRenewDate ) {
```

Comment:
Should the order of the checks be switched here, so that in the case where `purchase` is empty we don't try to pass it into `hasAmountAvailableToRefund()`?

You could do this either by just switching the order within the existing `if()` statement, or breaking it up into two separate `if()` statements each with its own early return.